UnityProject/Assets/Scripts/Systems/Explosions/NodeTypes/ExplosionNode.cs

```diff
@@ -68,57 +65,92 @@ public virtual void Process()
+
+			if (damageDealt > 0)
 			{
-				if (EffectName != null && EffectOverlayType != null && tileManager != null)
-				{
-					TimedEffect(v3int, DamageDealt * 10, EffectName, EffectOverlayType, tileManager);
-				}
-				EnergyExpended = DoDamage(matrix, DamageDealt, v3int);
+				//(Max): This is a terrible name. Whoever named it this way should be ashamed.
+				//I have no clue what's the context of this vector. Is it local position? Is it world position? Is it a direction? Who knows!
+				//Keep gatekeeping the codebase, it's not like there are other people working on this project..
+				var v3int = new Vector3Int(Location.x, Location.y, 0);
+				await ReguralProcessingToTilesOnly(damageDealt, v3int);
+				_ = DoDamageToObjects(matrix, damageDealt, v3int);
+			}
+		}
 
-				foreach (var line in PresentLines)
-				{
-					line.ExplosionStrength -= EnergyExpended * (line.ExplosionStrength / DamageDealt);
-				}
-				AngleAndIntensity = Vector2.zero;
+		protected async UniTask ReguralProcessingToTilesOnly(float damageDealt, Vector3Int v3int)
+		{
+			var tileManager = matrix.TileChangeManager;
+			if (EffectName != null && EffectOverlayType != null && tileManager != null)
+			{
+				await TimedEffect(v3int, damageDealt * 10, EffectName, EffectOverlayType, tileManager);
 			}
+			var metaTileMap = matrix.MetaTileMap;
+			var energyExpended = DoDamageToTiles(matrix, damageDealt, v3int, metaTileMap);
+			foreach (var line in PresentLines)
+			{
+				line.ExplosionStrength -= energyExpended * (line.ExplosionStrength / damageDealt);
+			}
+			AngleAndIntensity = Vector2.zero;
 		}
 
 		//method that, surprise, does damage to stuff on node's tile. override for custom behaviour. must return EnergyExpended value
-		public virtual float DoDamage(Matrix matrix, float damageDealt, Vector3Int v3int)
+		public virtual float DoDamageToTiles(Matrix matrix, float damageDealt, Vector3Int v3int, MetaTileMap metaTileMap)
 		{
-			var metaTileMap = matrix.MetaTileMap;
 			float energyExpended = metaTileMap.ApplyDamage(v3int, damageDealt,
 			MatrixManager.LocalToWorldInt(v3int, matrix.MatrixInfo), AttackType.Bomb);
 
 			DamageLayers(damageDealt, v3int);
+			ChangeNodeTemp(matrix, damageDealt, v3int);
+
+			return energyExpended;
+		}
 
+		public virtual async UniTask DoDamageToObjects(Matrix matrix, float damageDealt, Vector3Int v3int)
+		{
 			foreach (var integrity in matrix.Get<Integrity>(v3int, true))
 			{
-				//Throw items and Objects
-				integrity.GetComponent<UniversalObjectPhysics>()?.NewtonianNewtonPush(AngleAndIntensity.Rotate90(), AngleAndIntensity.magnitude * 0.1f , 1, 3,
-					BodyPartType.Chest, integrity.gameObject, 15);
+				// Incase of multiple explosions occuring at once (i.e: multiple Gibtonites)
+				// trycatch this to avoid trying to destroy stuff that is already destroyed by other damage sources.
+				try
+				{
+					integrity.Physics.NewtonianNewtonPush(AngleAndIntensity.Rotate90(), AngleAndIntensity.magnitude * 0.1f , 1, 3,
```

Comment:
I forgot to edit , make sure that it checks if it's intangible or not, so it doesn't destroy mapping items

Reply:
Etheral stuff are marked as indestructible by default, there's no point in adding another expensive component check.